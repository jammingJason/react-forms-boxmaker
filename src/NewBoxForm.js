import React, { useState } from 'react';

const NewBoxForm = ({ createBox }) => {
  //  set the initial data for the formData variable to '' for each input
  const INITIAL_DATA = {
    height: '',
    width: '',
    color: '',
  };
  const [formData, setFormData] = useState(INITIAL_DATA);
  // sets the data in the form to the formData variable
  // each time an item is selected or an input has been changed
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => {
      return { ...formData, [name]: value };
    });
  };
  // Hanles the form being submitted
  const handleSubmit = (e) => {
    e.preventDefault();
    createBox({ ...formData });
    setFormData(INITIAL_DATA);
  };

  //  returns the inputs and lables for each
  return (
    <form onSubmit={handleSubmit}>
      <hr />
      <h1>Create a New Box!</h1>
      <label htmlFor="height">Box Height: </label>
      <input
        type="text"
        name="height"
        id="height"
        value={formData.height}
        onChange={handleChange}
      />
      <br />
      <label htmlFor="width">Box Width: </label>
      <input
        type="text"
        name="width"
        id="width"
        value={formData.width}
        onChange={handleChange}
      />
      <br />
      <label htmlFor="color">Color: </label>

      <input
        type="color"
        name="color"
        id="color"
        value={formData.color}
        onChange={handleChange}
      />
      <br />
      <button>Create Box</button>
    </form>
  );
};

export default NewBoxForm;
