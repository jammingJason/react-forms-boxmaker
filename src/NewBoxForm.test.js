import { render, screen, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';
test('renders NewBoxForm', () => {
  render(<NewBoxForm />);
  //   const linkElement = screen.getByText(/learn react/i);
  //   expect(linkElement).toBeInTheDocument();
});

test('matches snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

const createBox = () => {};

it('should add new item', function () {
  const { queryByText, getByLabelText, getByRole } = render(
    <NewBoxForm createBox={createBox} />
  );
  const height = getByLabelText('Box Height:');
  const width = getByLabelText('Box Width:');
  const btn = queryByText('Create Box');
  //   expect(queryByText('X')).not.toBeInTheDocument();
  fireEvent.change(height, { target: { value: 100 } });
  fireEvent.change(width, { target: { value: 100 } });
  fireEvent.click(btn);
  expect(getByRole('button')).toBeInTheDocument();
});
