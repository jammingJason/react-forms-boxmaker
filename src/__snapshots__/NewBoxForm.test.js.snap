// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <hr />
    <h1>
      Create a New Box!
    </h1>
    <label
      for="height"
    >
      Box Height: 
    </label>
    <input
      id="height"
      name="height"
      type="text"
      value=""
    />
    <br />
    <label
      for="width"
    >
      Box Width: 
    </label>
    <input
      id="width"
      name="width"
      type="text"
      value=""
    />
    <br />
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value=""
    />
    <br />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
